kolibri/plugins/pdf_viewer/assets/src/utils/event_utils.js

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This file has been modified to adapt the component to the needs of Kolibri
+ * The original file is available at:
+ * https://github.com/mozilla/pdf.js/blob/v2.14.305/web/event_utils.js
+ */
+
+// Modified: Just the eventBus class is used
+
+/**
+ * Simple event bus for an application. Listeners are attached using the `on`
+ * and `off` methods. To raise an event, the `dispatch` method shall be used.
+ */
+class EventBus {
```

Comment:
I copied this class to handle event based communication similar to how they do in mozilla's pdf reader. I think it could be replaced with native javascript custom events, although I'm not sure how much it would affect cross-browser compatibility.

Reply:
As I mentioned during our checkin yesterday, I chatted with @rtibbles about the eventBus, and he suggested following [this `_eventDispatch` pattern that we use in the Kolibri Design System](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/utils/pluginMediator.js#L64). 

As we are using it for the design system, I think it should be fine for all of the browsers we support. 

You can look through this file to see some examples of how it is being used. Take a look and see what you think! Let me or @rtibbles know if you have any questions.

Reply:
My main point here was that you can just use an instance of the `Vue` object as an event bus, which is how it is being used in the KDS code that @marcellamaki linked. So this should reduce the amount of vendored code required, and mean that we can rely on Vue's event bus which we are pretty darn sure works!

Reply:
Nice! I'll use that then